### PR TITLE
sw_engine common: Added neon version of rasterRGBA32 API.

### PR DESCRIPTION
**Pull request by mihashco**
Changes:
- Added 'neon' vector option in build system
- Introduced neon version of rasterRGBA32 API, which improves speed of the function on ARM cpu's around ~35% (measured on 800x800 array of uint32_t)

### CODE REVIEW

**Review by mmaciola — APPROVED** [review]

**Comment by hermet** [comment]
@mihashco did you test on real device?

File: src/lib/sw_engine/tvgSwCommon.h [hunk 1]
```diff
@@ -383,6 +387,22 @@ static inline void rasterRGBA32(uint32_t *dst, uint32_t val, uint32_t offset, in
...
+#elif defined(THORVG_NEON_VECTOR_SUPPORT)
+    uint32_t iterations = len / 4;
+    uint32_t neon_filled = iterations * 4;
+    int32_t left = 0;
+
+    uint32_t i = 0;
+    dst+=offset;
+    uint32x4_t v_val = { val, val, val, val };
+
+    for (i = 0; i < iterations; ++i) {
```
Comment (hermet) [inline src/lib/sw_engine/tvgSwCommon.h:399]: please fix convention:
for (uint32_t i = 0; ...)

File: src/lib/sw_engine/tvgSwCommon.h [hunk 2]
```diff
@@ -383,6 +387,22 @@ static inline void rasterRGBA32(uint32_t *dst, uint32_t val, uint32_t offset, in
     dst+= avxFilled;
 
     while (leftovers--) *dst++ = val;
+#elif defined(THORVG_NEON_VECTOR_SUPPORT)
+    uint32_t iterations = len / 4;
+    uint32_t neon_filled = iterations * 4;
```
Comment (hermet) [inline src/lib/sw_engine/tvgSwCommon.h:392]: please fix convention:
neon_filled => neonFilled

File: src/lib/sw_engine/tvgSwCommon.h [hunk 3]
```diff
@@ -383,6 +387,22 @@ static inline void rasterRGBA32(uint32_t *dst, uint32_t val, uint32_t offset, in
...
+    dst+=offset;
+    uint32x4_t v_val = { val, val, val, val };
+
+    for (i = 0; i < iterations; ++i) {
+        vst1q_u32(dst, v_val);
+        dst += 4;
+    }
+
+    left = len - neon_filled;
+    while(left--) *dst++ = val;
```
Comment (hermet) [inline src/lib/sw_engine/tvgSwCommon.h:405]: please fix convention:
while^(left

**Review by hermet — CHANGES_REQUESTED** [review]
Please fix nitpicks :+1:

**Comment by mihashco** [comment]
> @mihashco did you test on the real device?

yes, I tested this function and checked its result on arm. Also thorVG with this funciton was tested on ARM.

**Review by hermet — APPROVED** [review]

**Pull request merged**